addons/web/static/src/core/name_service.js

```diff
@@ -0,0 +1,114 @@
+    dependencies: ["orm"],
+    async: ["loadDisplayNames"],
+    start(env, { orm }) {
+        let cache = {};
+        const batches = {};
+
+        function clearCache() {
+            cache = {};
+        }
+
+        env.bus.addEventListener("ACTION_MANAGER:UPDATE", clearCache);
+
+        /**
+         * @param {string} resModel valid resModel name
+         * @param {DisplayNames} displayNames
+         */
+        function addDisplayNames(resModel, displayNames) {
+            if (!cache[resModel]) {
+                cache[resModel] = {};
+            }
+            const mapping = cache[resModel];
+            for (const resId in displayNames) {
+                mapping[resId] = new Deferred();
+                mapping[resId].resolve(displayNames[resId]);
+            }
+        }
+
+        /**
+         * @param {string} resModel valid resModel name
+         * @param {number[]} resIds valid ids
+         * @returns {Promise<DisplayNames>}
+         */
+        async function loadDisplayNames(resModel, resIds) {
+            if (!cache[resModel]) {
+                cache[resModel] = {};
+            }
+            const mapping = cache[resModel];
+            const proms = [];
+            const resIdsToFetch = [];
+            for (const resId of unique(resIds)) {
+                if (!(resId in mapping)) {
+                    mapping[resId] = new Deferred();
+                    if (isId(resId)) {
+                        resIdsToFetch.push(resId);
+                    } else {
+                        mapping[resId].resolve(ERROR_INVALID);
+                    }
+                }
+                proms.push(mapping[resId]);
+            }
+            if (resIdsToFetch.length) {
+                if (batches[resModel]) {
+                    batches[resModel] = unique([...batches[resModel], ...resIdsToFetch]);
+                } else {
+                    batches[resModel] = resIdsToFetch;
+                    await Promise.resolve();
+                    const idsInBatch = batches[resModel];
+                    delete batches[resModel];
+                    orm.webSearchRead(resModel, [["id", "in", idsInBatch]], ["display_name"]).then(
+                        ({ records }) => {
+                            const displayNames = Object.fromEntries(
+                                records.map((rec) => [rec.id, rec.display_name])
+                            );
+                            const missingIds = [];
+                            for (const resId of idsInBatch) {
+                                if (resId in displayNames) {
+                                    mapping[resId].resolve(displayNames[resId]);
+                                } else {
+                                    missingIds.push(resId);
+                                }
+                            }
+                            if (missingIds.length) {
+                                orm.call(resModel, "exists", missingIds).then((existing) => {
```

Comment:
`args` is supposed to be a list and therefore ids should be a list in a list, right ?
```suggestion
                                orm.call(resModel, "exists", [missingIds]).then((existing) => {
```